src/ImageSharp/Formats/Qoi/QoiEncoderCore.cs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Qoi;
+
+/// <summary>
+/// Image encoder for writing an image to a stream as a QOi image
+/// </summary>
+public class QoiEncoderCore : IImageEncoderInternals
+{
+    /// <summary>
+    /// The encoder with options
+    /// </summary>
+    private readonly QoiEncoder encoder;
+
+    /// <summary>
+    /// Used the manage memory allocations.
+    /// </summary>
+    private readonly MemoryAllocator memoryAllocator;
+
+    /// <summary>
+    /// The configuration instance for the encoding operation.
+    /// </summary>
+    private readonly Configuration configuration;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QoiEncoderCore"/> class.
+    /// </summary>
+    /// <param name="encoder">The encoder with options.</param>
+    /// <param name="memoryAllocator">The <see cref="MemoryAllocator" /> to use for buffer allocations.</param>
+    public QoiEncoderCore(QoiEncoder encoder, MemoryAllocator memoryAllocator, Configuration configuration)
```

Comment:
If you run a release build locally, it should catch these:

https://github.com/SixLabors/ImageSharp/actions/runs/5449320272/jobs/9936168476?pr=2446#step:13:67

Reply:
Ooh yeah, I forgot to check StyleCop on Release